### PR TITLE
Upgrade grpc and abseil for SPM and cocoapods

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: FirebaseFirestore.podspec [hunk 1]
```diff
@@ -98,7 +98,7 @@ Google Cloud Firestore is a NoSQL document database built for automatic scaling,
   s.dependency 'abseil/time', abseil_version
   s.dependency 'abseil/types', abseil_version
 
-  s.dependency 'gRPC-C++', '~> 1.28.0'
+  s.dependency 'gRPC-C++', '~> 1.44.0'
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply FirebaseFirestore.podspec]: Done.

**Comment by dennycd** [comment]
abseil build break to be fixed via https://github.com/firebase/abseil-cpp-SwiftPM/pull/9

**Review by paulb777 — COMMENTED** [review]
- Why aren't the abseil versions consistent between CocoaPods and SPM?
- Why is the CocoaPods abseil version locked to a patch release but SPM enables patch updates?
- Why do we allow patch updates for abseil but not gRPC
- Is the CI build warning in abseil fixable: 
-     - WARN  | xcodebuild:  /Users/runner/Library/Developer/Xcode/DerivedData/App-ffzsphyzsoubipeuqfnlcfzcgour/Build/Products/Release-iphonesimulator/abseil/absl.framework/Headers/strings/str_split.h:517:7: warning: Forwarding reference passed to std::move(), which may unexpectedly cause lvalues to be moved; use std::forward() instead [bugprone-move-forwarding-reference]
    - WARN  | xcodebuild:  /Users/runner/Library/Developer/Xcode/DerivedData/App-ffzsphyzsoubipeuqfnlcfzcgour/Build/Products/Release-iphonesimulator/abseil/absl.framework/Headers/strings/str_split.h:542:7: warning: Forwarding reference passed to std::move(), which may unexpectedly cause lvalues to be moved; use std::forward() instead [bugprone-move-forwarding-reference]

**Comment by wu-hui** [comment]
> * Why aren't the abseil versions consistent between CocoaPods and SPM?
> * Why is the CocoaPods abseil version locked to a patch release but SPM enables patch updates?
> * Why do we allow patch updates for abseil but not gRPC
> * Is the CI build warning in abseil fixable:
> * ```
>   - WARN  | xcodebuild:  /Users/runner/Library/Developer/Xcode/DerivedData/App-ffzsphyzsoubipeuqfnlcfzcgour/Build/Products/Release-iphonesimulator/abseil/absl.framework/Headers/strings/str_split.h:517:7: warning: Forwarding reference passed to std::move(), which may unexpectedly cause lvalues to be moved; use std::forward() instead [bugprone-move-forwarding-reference]
>   ```
>   
>   
>       
>         
>       
>   
>         
>       
>   
>       
>     
>   
>   * WARN  | xcodebuild:  /Users/runner/Library/Developer/Xcode/DerivedData/App-ffzsphyzsoubipeuqfnlcfzcgour/Build/Products/Release-iphonesimulator/abseil/absl.framework/Headers/strings/str_split.h:542:7: warning: Forwarding reference passed to std::move(), which may unexpectedly cause lvalues to be moved; use std::forward() instead [bugprone-move-forwarding-reference]

> * Why aren't the abseil versions consistent between CocoaPods and SPM?
Abseil on cocoapods is behind, I am using the latest I can use. @dennycd can you help?

> * Why is the CocoaPods abseil version locked to a patch release but SPM enables patch updates?
> * Why do we allow patch updates for abseil but not gRPC
I am not sure why abseil is fixed on version. The history is lost..I am guess we have been bitten before by absl. I am enabling it for now...let's see what happens.

gRPC being fixed is a mistake.
> * Is the CI build warning in abseil fixable:
Will check.

**Comment by dennycd** [comment]
> Abseil on cocoapods is behind, I am using the latest I can use. @dennycd can you help?
> 
> >

looks like abseil-cpp-SwiftPM's  tag ([0.20220203.0](https://github.com/firebase/abseil-cpp-SwiftPM/releases/tag/0.20220203.0) and up) are indeed using the same abseil version as latest cocoapod (20211102.0), just that the tag names from abseil-cpp-SwiftPM doesn't exactly match abseil's release tags. 

We can either keep the current tag scheme for now, or re-tag abseil-cpp-SwiftPM to match that from cocoapod (e.g. 0.20211102.0).   Hopefully once we resolve abseil SPM & Cocoapod dependency w/ abseil team, we can start using a consistent release tag from the same repo

**Comment by morganchen12** [comment]
This PR also includes https://github.com/grpc/grpc/pull/27300, which resolves https://github.com/firebase/quickstart-ios/issues/1099.

**Review by paulb777 — APPROVED** [review]
LGTM on green

**Pull request merged**